crates/wasm/src/view_server.rs

```diff
@@ -101,20 +101,19 @@ impl ViewServer {
     }
 
     /// Scans block for notes, swaps
-    /// This method does not return SCT updates (nct_updates).
-    /// Although we can make it do if we want to save SCT updates after each blob is processed
-    /// Arguments:
+    /// Returns true if the block contains new notes, swaps or false if the block is empty for us
     ///     compact_block: `v1alpha1::CompactBlock`
+    /// Scan results are saved in-memory rather than returned
+    /// Use `get_updates()` to get the scan results
     /// Returns: `ScanBlockResult`
     #[wasm_bindgen]
-    pub async fn scan_block(&mut self, compact_block: JsValue) -> Result<JsValue, Error> {
+    pub async fn scan_block(&mut self, compact_block: JsValue) -> Result<bool, Error> {
         let result = self.scan_block_inner(compact_block).await?;
```

Comment:
Didn't we get rid of this need for `_inner` functions? Would be ideal if we could fix that here too.

Reply:
Some large _inner functions remain, I'll fix them